File: taskhive-backend/src/main/java/com/digiwork/taskhive/module/analytics/dto/AdminDashboardResponse.java
package com.digiwork.taskhive.module.analytics.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminDashboardResponse {

    private long totalTasks;
    private long activeTasks;
    private long overdueTasks;
    private long completedTasks;
    private BigDecimal completionRate;
    private int totalEmployees;
    private BigDecimal avgCompletionHours;
    private LocalDate metricDate;
}

File: taskhive_flutter/lib/features/analytics/data/models/admin_dashboard_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'admin_dashboard_response.freezed.dart';
part 'admin_dashboard_response.g.dart';

@freezed
class AdminDashboardResponse with _$AdminDashboardResponse {
  const factory AdminDashboardResponse({
    int? totalTasks,
    int? activeTasks,
    int? overdueTasks,
    int? completedTasks,
    double? completionRate,
    int? totalEmployees,
    double? avgCompletionHours,
    String? metricDate,
  }) = _AdminDashboardResponse;

  factory AdminDashboardResponse.fromJson(Map<String, dynamic> json) =>
      _$AdminDashboardResponseFromJson(json);
}
